File: src/types/pos.ts
export interface PosSession {
  id: string;
  register_id: string;
  cashier_id: string;
  opening_fund: number;
  closing_fund: number | null;
  status: "open" | "closed";
  ticket_count: number;
  total_sales: number;
  opened_at: string;
  closed_at: string | null;
}

export interface PosTicket {
  id: string;
  session_id: string;
  ticket_number: number;
  status: "active" | "held" | "paid" | "cancelled";
  total_ht: number;
  total_tax: number;
  total_ttc: number;
  payment_status: "pending" | "partial" | "paid";
  created_at: string;
}

export interface TicketLine {
  id: string;
  ticket_id: string;
  article_id: string;
  article_name: string;
  quantity: number;
  unit_price: number;
  tax_rate: number;
  total_ht: number;
  total_ttc: number;
}

export interface PaymentLine {
  id: string;
  ticket_id: string;
  payment_mode: string;
  amount: number;
}

export type CashMovementType = "in" | "out";

export interface CashMovement {
  id: string;
  session_id: string;
  movement_type: CashMovementType;
  amount: number;
  description: string;
  user_id: string;
  user_name: string;
  created_at: string;
}

export interface CreateCashMovement {
  session_id: string;
  movement_type: CashMovementType;
  amount: number;
  description: string;
  user_id: string;
  user_name: string;
}

export interface CashMovementSummary {
  opening_fund: number;
  total_in: number;
  total_out: number;
  current_cash: number;
}

export interface CashSessionTotals {
  session_id: string;
  register_id: string;
  cashier_id: string;
  status: "open" | "closed" | string;
  opening_fund: number;
  closing_fund: number | null;
  total_in: number;
  total_out: number;
  theoretical_closing: number;
  variance: number;
  opened_at: string;
  closed_at: string | null;
}

File: src-tauri/src/domain/pos.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TicketLine {
    pub id: String,
    pub ticket_id: String,
    pub article_id: String,
    pub article_name: String,
    pub quantity: i64,
    pub unit_price: i64,
    pub tax_rate: i64,
    pub total_ht: i64,
    pub total_ttc: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PaymentLine {
    pub id: String,
    pub ticket_id: String,
    pub payment_mode: String,
    pub amount: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct PosSession {
    pub id: String,
    pub register_id: String,
    pub cashier_id: String,
    pub opening_fund: i64,
    pub closing_fund: Option<i64>,
    pub status: String,
    pub ticket_count: i64,
    pub total_sales: i64,
    pub opened_at: DateTime<Utc>,
    pub closed_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct PosTicket {
    pub id: String,
    pub session_id: String,
    pub ticket_number: i64,
    pub status: String,
    pub total_ht: i64,
    pub total_tax: i64,
    pub total_ttc: i64,
    pub payment_status: String,
    pub created_at: DateTime<Utc>,
}

impl PosSession {
    pub fn open(register_id: &str, cashier_id: &str, opening_fund: i64) -> Self {
        Self {
            id: Uuid::new_v4().to_string(),
            register_id: register_id.to_string(),
            cashier_id: cashier_id.to_string(),
            opening_fund,
            closing_fund: None,
            status: "open".to_string(),
            ticket_count: 0,
            total_sales: 0,
            opened_at: Utc::now(),
            closed_at: None,
        }
    }
}
